src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleDamageParticles.kt

```diff
@@ -0,0 +1,171 @@
+/*
+ * This file is part of LiquidBounce (https://github.com/CCBlueX/LiquidBounce)
+ *
+ * Copyright (c) 2015 - 2024 CCBlueX
+ *
+ * LiquidBounce is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * LiquidBounce is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with LiquidBounce. If not, see <https://www.gnu.org/licenses/>.
+ */
+package net.ccbluex.liquidbounce.features.module.modules.render
+
+import net.ccbluex.liquidbounce.config.Choice
+import net.ccbluex.liquidbounce.event.events.OverlayRenderEvent
+import net.ccbluex.liquidbounce.event.handler
+import net.ccbluex.liquidbounce.event.repeatable
+import net.ccbluex.liquidbounce.features.module.Category
+import net.ccbluex.liquidbounce.features.module.Module
+import net.ccbluex.liquidbounce.render.Fonts
+import net.ccbluex.liquidbounce.render.engine.Color4b
+import net.ccbluex.liquidbounce.render.renderEnvironmentForGUI
+import net.ccbluex.liquidbounce.utils.entity.box
+import net.ccbluex.liquidbounce.utils.entity.eyes
+import net.ccbluex.liquidbounce.utils.render.WorldToScreen
+import net.minecraft.entity.LivingEntity
+import net.minecraft.util.math.Vec3d
+import kotlin.math.PI
+import kotlin.math.abs
+import kotlin.math.pow
+import kotlin.math.sin
+
+/**
+ * DamageParticles module
+ *
+ * Show health changes of entities
+ */
+object ModuleDamageParticles : Module("DamageParticles", Category.RENDER) {
+
+    private val scale by float("Scale", 1.5F, 0.25F..4F)
+    private val ttl by float("TimeToLive", 1.5F, 0.5F..5.0F, "s")
+    private val transitionY by float("TransitionY", 1.0F, -2.0F..2.0F)
+    private val transitionType = choices(
+        "TransitionType", Linear, arrayOf(
+            Linear, Sine, Exponential, EaseInOut, Cubic, Elastic, Back, Step
+        )
+    )
+
+    private fun Double.doTransition() = transitionType.activeChoice(this)
+
+    private val healthMap = hashMapOf<LivingEntity, Float>()
```

Comment:
Perhaps you could use an Object2Float map.

Reply:
There are several implementations of Object2FloatMap, I choosed one of them without considerations of performance.